src/CattyTests/Extensions/ProjectDetailStoreViewControllerExtensionTests.swift

```diff
@@ -0,0 +1,50 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+@testable import Pocket_Code
+import XCTest
+
+class ProjectDetailStoreViewControllerExtensionTests: XCTestCase {
+    var expectedZippedProjectData: Data!
+    var projectDetailStoreVC: ProjectDetailStoreViewController!
+    var cbFileManager: CBFileManagerMock!
+
+    override func setUp() {
+        self.projectDetailStoreVC = ProjectDetailStoreViewController.init()
+        self.projectDetailStoreVC.project = CatrobatProject.init()
+        self.projectDetailStoreVC.project.projectID = "817"
+        self.projectDetailStoreVC.project.projectName = "Tic-Tac-Toe Master"
+        self.projectDetailStoreVC.project.downloadUrl = "https://web-test.catrob.at/pocketcode/download/817.catrobat"
+
+        self.expectedZippedProjectData = "zippedProjectData".data(using: .utf8)
+        self.projectDetailStoreVC.viewDidLoad()
+        self.cbFileManager = CBFileManagerMock(zipData: expectedZippedProjectData)
+    }
+
+    func testStoreProjectDownloader() {
+        let projectName = "Tic-Tac-Toe Master"
+
+        XCTAssertNotNil(self.projectDetailStoreVC.storeProjectDownloader)
+        self.projectDetailStoreVC.download(name: projectName)
```

Comment:
Is this test making actual http requests?

Reply:
Please revise this test, is it really testing what it should?

Reply:
Changed it, thank you!